riptide-stream/src/main/java/org/zalando/riptide/stream/StreamConverter.java

```diff
@@ -0,0 +1,79 @@
+package org.zalando.riptide.stream;
+
+/*
+ * ⁣​
+ * Riptide: Stream
+ * ⁣⁣
+ * Copyright (C) 2015 - 2016 Zalando SE
+ * ⁣⁣
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ​⁣
+ */
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Type;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.springframework.http.HttpInputMessage;
+import org.springframework.http.MediaType;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+class StreamConverter extends MappingJackson2HttpMessageConverter {
```

Comment:
My ideas to improve this class:
- don't extend MappingJackson2HttpMessageConverter, if you need it, use an instance field (maybe static?)
- don't support List/Array directly, there is no need since clients can just use MappingJackson2HttpMessageConverter and immediately call `List.stream()` on the return value
- only support two concrete media types: `application/json-seq` and `application/x-json-stream`
- don't try to compete with MappingJackson2HttpMessageConverter, but work in conjunction